core/apple/src/BuffersApple.kt

```diff
@@ -0,0 +1,67 @@
+@file:OptIn(UnsafeNumber::class)
+
+package kotlinx.io
+
+import kotlinx.cinterop.*
+import kotlinx.io.bytestring.ByteString
+import kotlinx.io.bytestring.buildByteString
+import platform.Foundation.*
+import platform.darwin.NSUInteger
+import platform.darwin.NSUIntegerMax
+import platform.posix.memcpy
+import platform.posix.uint8_tVar
+
+internal fun Buffer.write(source: CPointer<uint8_tVar>?, maxLength: Int) {
```

Comment:
Please, verify that `maxLength` is non-negative and that `source` is not null.

Reply:
I made `source` non-optional here and did the check in the caller and also applied these changes to `Buffer.readAtMost()`.